src/nools/target-emitter.js

```diff
@@ -0,0 +1,72 @@
+var prepareDefinition = require('./definition-preparation');
+
+function targetEmitter(targets, c, Utils, Target, emit) {
+  for (var idx1 = 0; idx1 < targets.length; ++idx1) {
+    var target = targets[idx1];
+    prepareDefinition(target);
+
+    switch (target.appliesTo) {
+      case 'contacts':
+        emitTargetFor(target, c, undefined, Target, Utils, emit);
```

Comment:
Can we switch `c` and `r` to be at the end, so `r` can be optional?

Reply:
Yep - good call